app/helpers/reactjs_helper.rb

```diff
@@ -0,0 +1,8 @@
+module ReactjsHelper
+
+  def mount_react_component(name, selector, data)
```

Comment:
Oh, I completely missed the point. I'm not worried about conflicts. If there's a risk, it's worthwhile considering refactoring the markup.

In this particular case, it wouldn't make much of a difference but I think it could be problematic as a general solution, although perhaps the helper could be expanded to handle some of the problems.

- we won't always want a div as the container element (li, is likely)
- the page itself could need to control styling of the container (at least layout)
- if the container is not the last element on the page, javascript will be injected into the markup body, or in several different places if there is more than one element mounted on the page (inspection via browser tools could be more difficult)
- in the case of a more generic component name, the id wouldn't be very semantic (chart-923, chart-8501, etc)

This may depend a lot on the code someone is used to looking at. I wouldn't find the proposed solution easy to work with but my background isn't representative of the team.